### PR TITLE
[Docs] file upload exercises and second correction in exam mode

### PR DESCRIPTION
**Pull request by SimonEntholzer**
### Motivation and Context

I added missing documentation for the second correction of exams, and documentation for file-upload exercises.

### Steps for Testing

easiest way to "test" is, to check out the repository locally. Then build the docs. See the README in the docs folder.

### CODE REVIEW

**Review by philgabr — CHANGES_REQUESTED** [review]
Looks good.

However, it seems like the submissions.png is saved in the wrong folder (.../exams/instructor instead of .../exams/instructor/buttons) and therefore can not be displayed/found:
image

**Comment by SimonEntholzer** [comment]
> Looks good.
> 
> However, it seems like the submissions.png is saved in the wrong folder (.../exams/instructor instead of .../exams/instructor/buttons) and therefore can not be displayed/found:
> image

Yea, I somehow missed to add the file to git, now it is visible. :)

File: docs/user/exercises/file-upload.rst [hunk 1]
```diff
@@ -1,2 +1,75 @@
...
+
+.. contents:: Content of this document
+    :local:
+    :depth: 2
+
+
+Overview
+--------
+
+Conducting a file exercise consists of 3 steps distributed among
```
Comment (FrankeLukas) [inline docs/user/exercises/file-upload.rst:12]: file upload exercise?

File: docs/user/exercises/file-upload.rst [hunk 2]
```diff
@@ -1,2 +1,75 @@
...
+instructor, Artemis and students:
```
Comment (FrankeLukas) [inline docs/user/exercises/file-upload.rst:13]: The wording here is odd imo. Why not just `Conducting a file exercise consists of 3 steps:`?

File: docs/user/exercises/file-upload.rst [hunk 3]
```diff
@@ -1,2 +1,75 @@
...
+Assessment
+----------
+
+- When the due date is over you can assess the submissions. From the assessment dashboard go to exercise assessment dashboard of the file-upload exercise.
+- There you can assess the submitted student submissions, by first downloading the file, and then creating feedback with points.
+
+.. |edit| image:: programming/edit.png
+.. |course-management| image:: programming/course-management.png
+.. |save| image:: file-upload/save_button.png
+.. |submit| image:: file-upload/submit.png
```
Comment (FrankeLukas) [inline docs/user/exercises/file-upload.rst:72]: This file does not seem to exist?

File: docs/user/exercises/file-upload.rst [hunk 4]
```diff
@@ -1,2 +1,75 @@
...
+
+- After a file was uploaded it can then be downloaded again with the link below.
+
+    .. figure:: file-upload/file_upload_exercise_student_submitted.png
+              :align: center
+
+Assessment
+----------
+
+- When the due date is over you can assess the submissions. From the assessment dashboard go to exercise assessment dashboard of the file-upload exercise.
```
Comment (FrankeLukas) [inline docs/user/exercises/file-upload.rst:66]: `file-upload` -> `file upload` to be consistent with the rest of the document.

File: docs/user/exercises/file-upload.rst [hunk 5]
```diff
@@ -1,2 +1,75 @@
...
+- There you can assess the submitted student submissions, by first downloading the file, and then creating feedback with points.
+
+.. |edit| image:: programming/edit.png
+.. |course-management| image:: programming/course-management.png
+.. |save| image:: file-upload/save_button.png
+.. |submit| image:: file-upload/submit.png
+    :scale: 50
+.. |browse| image:: file-upload/browse.png
```
Comment (FrankeLukas) [inline docs/user/exercises/file-upload.rst:74]: This file also appears to be missing?

**Review by FrankeLukas — CHANGES_REQUESTED** [review]

File: docs/user/exercises/file-upload.rst (hunk 1 above)
Comment (SimonEntholzer) [reply docs/user/exercises/file-upload.rst:12]:
```suggestion
Conducting a file upload exercise consists of 3 steps distributed among
```

File: docs/user/exercises/file-upload.rst (hunk 2 above)
Comment (SimonEntholzer) [reply docs/user/exercises/file-upload.rst:13]: Yep that sounds more concise, I will use your wording :)

File: docs/user/exercises/file-upload.rst (hunk 3 above)
Comment (SimonEntholzer) [reply docs/user/exercises/file-upload.rst:72]: jep I just realized I missed to add some more files to git

File: docs/user/exercises/file-upload.rst (hunk 5 above)
Comment (SimonEntholzer) [reply docs/user/exercises/file-upload.rst:74]: same issue here

**Review by philgabr — APPROVED** [review]
Approve :+1:

**Review by JohannesWeiss — APPROVED** [review]
Changes look good to me.

**Review by krusche — APPROVED** [review]

**Pull request merged**